config/main.py

```diff
@@ -1393,8 +1393,8 @@ def multiasic_write_to_db(filename, load_sysinfo):
 def config_file_yang_validation(filename):
     config = read_json_file(filename)
 
-    # Check if the config is empty
-    if not config or not isinstance(config, dict):
+    # Check if the config is not a dictionary
+    if not isinstance(config, dict):
```

Comment:
In line 1416 it will fail when config_to_check is None.
Also it will fail in line 2027:
https://github.com/sonic-net/sonic-utilities/blob/bf7bd4a655ee7be1feaf2d81c41336c55009b19b/config/main.py#L2077

How about add check before:
line 1416
```

   for scope in asic_list:
        config_to_check = config.get(scope) if multi_asic.is_multi_asic() else config
        if not config_to_check:
               try:
```
line 2027
```
for scope in asic_list:
       host_config = config_to_check.get(scope) if multi_asic.is_multi_asic() else config_to_check
       if not host_config:
            table_hard_dependency_check(host_config)
```